sourcegraph/learn
# Validate collections yaml

### What should this PR do?
Resolves [DEVED-22](https://sourcegraph.atlassian.net/browse/DEVED-22) by adding validation for our `collections.yaml`.

Refactors `loadCollections` as part of this, also with a future eye to enabling content type checks as part of the validation flow (i.e. if no collection of `type = recordType` then skip).

Also prefers `null` to undefined and setting empty strings as default values.

### Why are we making this change?
We would like to be able to validate our data, ensuring it matches the intended format.

### What are the acceptance criteria? 
- Validations should work for record collections.
- Validations should work for author collections.
- Site should work as expected.

### How should this PR be tested?
- Check out branch, ensure site is working as expected.
- Ensure no new errors on linting.

## Pull request process

**Reviewers**:

1. Test functionality using the criteria above.
2. Offer tips for efficiency, feedback on best practices, and possible alternative approaches and things that may not have been considered.
3. For shorter, "quick" PRs, use your best judgement on #​2.
4. Use a collaborative approach and provide resources and/or context where appropriate.
5. Provide screenshots/grabs where appropriate to show findings during review.

**Reviewees**:

1. Prefer incremental and appropriately-scoped changes.
2. Leave a comment on things you want explicit feedback on.
3. Respond clearly to comments and questions.
